akka-stream/src/main/scala/akka/stream/scaladsl/Sink.scala

```diff
@@ -75,8 +76,22 @@ object Sink {
 
   /**
    * A `Sink` that materializes into a `Future` of the first value received.
+   * If the stream completes before signaling at least a single element, the Future will be failed with an [[NoSuchElementException]].
+   * If the stream signals an error errors before signaling at least a single element, the Future will be failed with the streams exception.
+   *
+   * See also [[headOption]].
+   */
+  def head[T]: Sink[T, Future[T]] = new Sink[T, Future[Option[T]]](new HeadOptionSink[T](DefaultAttributes.headSink, shape("HeadSink")))
```

Comment:
That should be solved then in other ways. If naming would force us to *never reuse* any other DSL element to build a new one, then that would be sad :(